java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java

```diff
@@ -180,4 +203,109 @@ private void uiTest() throws Exception {
       screens.exception(e);
     }
   }
+
+  private static class IllegalTxSignTestFileFormatException extends Exception {
+    public IllegalTxSignTestFileFormatException(String msg) {
+      super(msg);
+    }
+  }
+
+  private void signTxTest() throws Exception {
+
+    // TODO: implement HSM test
+    if (nCipher) {
+      System.out.println("Transaction signing regression test not yet implemented for nCipher");
+      return;
+    }
+
+    // Passed and failed test cases, for valid and invalid test vectors.
+    // For a valid test vector, test passes (ok) if and only if subzero response matches
+    // expected response.
+    // For an invalid test vector, test passes (ok) if and only if subzero response does not
```

Comment:
had a little bit of trouble understanding "invalid" test vectors

Reply:
I will try clarifying a little bit further in the comment in the upcoming commit.